mrTrix3 compatibility

garikoitz opened the pull request:
1.- In some cases changes are just for visualization (each parameter in a new line), but usually I add a new parameter (mrTrixVersion).
2.- I found a couple of bugs with mrTrix (not the correct parameters were being passed. 
3.- I had to make changes in file/path names as well, it wasn't working for me, hopefully it would be ok, please check it. 
4.- I had to create a copy of most of the vistasoft/mrtrix, maybe it will be easier to change them in Vistasoft and do a pull_request there, please let me know. 
5. mrTrix3 calls are quite different from mrTrix2, I code it in a way that the mrTrix2 calls were unaltered, but I didn't test them since I don't have mrTrix2 installed. 
6.- WARNING: the results from mrTrix2 and mrTrix3 are not compatible.

jyeatman commented:
Gari, thanks for the PR. @dstrodtman lets go through this together and test it on your data

dstrodtman commented on functions/AFQ_Create.m:
```diff
@@ -268,7 +273,10 @@
         if ~exist(mrtrixdir,'dir'),mkdir(mrtrixdir);end
         % Get the lmax from the afq structure
         lmax = AFQ_get(afq,'lmax');
-        files = mrtrix_init(AFQ_get(afq, 'dt6 path',ii),lmax,mrtrixdir);
```
Shouldn't we always get the version with AFQ_get

dstrodtman commented on functions/AFQ_SegmentFiberGroups.m:
```diff
@@ -156,8 +156,17 @@
 % Set the directory where templates can be found
 tdir = fullfile(fileparts(which('mrDiffusion.m')), 'templates');
 % Initialize spm defualts for normalization
-spm_defaults; global defaults; params = defaults.normalise.estimate;
```
Can these changes be split into another PR? I will check separately

garikoitz replied:
If it is working for you just ignore the change please. First of all I wanted to make sure that it was working in my system and I had to do some changes. This was tested on OSX and now I am moving everything to the cluster with Red Hat 6.

dstrodtman commented on utilities/AFQ_mrtrixInit.m:
```diff
@@ -32,42 +32,65 @@
 % http://www.brain.org.au/software/mrtrix/tractography/preprocess.html
 % 
-
-if notDefined('mrtrix_folder'), mrtrix_folder = 'mrtrix'; end
```
I think we should keep the same directory structure by default

garikoitz replied:
It didn't work for me and I did a quick fix to make it work. If it is working for you please don't accept the change. Please notice that structure is different to the one found in vistasoft.

jyeatman commented on utilities/AFQ_mrtrix_track.m:
```diff
@@ -0,0 +1,168 @@
+switch mode
+  case {'prob','probabilistic tractography'}
+    mode_str2 = 'SD_PROB';
+    mode_str3 = 'Tensor_Prob';
```
This should be SD_Prob I think. We don't want to track based on the tensor. We want to track based on SD

jyeatman replied:
Maybe we should support all these options
FACT:Mori, S.; Crain, B. J.; Chacko, V. P. & van Zijl, P. C. M. Three-dimensional tracking of axonal projections in the brain by magnetic resonance imaging. Annals of Neurology, 1999, 45, 265-269
iFOD1 or SD_STREAM:Tournier, J.-D.; Calamante, F. & Connelly, A. MRtrix: Diffusion tractography in crossing fiber regions. Int. J. Imaging Syst. Technol., 2012, 22, 53-66
iFOD2:Tournier, J.-D.; Calamante, F. & Connelly, A. Improved probabilistic streamlines tractography by 2nd order integration over fibre orientation distributions. Proceedings of the International Society for Magnetic Resonance in Medicine, 2010, 1670
Nulldist1 / Nulldist2:Morris, D. M.; Embleton, K. V. & Parker, G. J. Probabilistic fibre tracking: Differentiation of connections from chance events. NeuroImage, 2008, 42, 1329-1339
Tensor_Det:Basser, P. J.; Pajevic, S.; Pierpaoli, C.; Duda, J. & Aldroubi, A. In vivo fiber tractography using DT-MRI data. Magnetic Resonance in Medicine, 2000, 44, 625-632
Tensor_Prob:Jones, D. Tractography Gone Wild: Probabilistic Fibre Tracking Using the Wild Bootstrap With Diffusion Tensor MRI. IEEE Transactions on Medical Imaging, 2008, 27, 1268-1274

garikoitz replied:
You are right, the first one is an error. I will add the rest of the algo-s to the call.

jyeatman commented on utilities/check_mrTrix.m:
```diff
@@ -1,17 +1,38 @@
-function [status, mrTrixpaths] = check_mrTrix
```
Put an argument check in here so that it checks mrtrix 2 by default if there is no input argument

jyeatman commented on utilities/check_mrTrix_Version.m:
```diff
@@ -0,0 +1,19 @@
+function mrTrixVersion = check_mrTrix_Version
```
This function needs to call AFQ_mrtrix_set_ld_path. Otherwise mrview will error because of libraries

garikoitz replied:
Yes, in OSX it's OK but just have had the same problem in linux.

dstrodtman replied:
This doesn't look like it has been fixed. For this function to run properly we need to first call AFQ_mrtrix_set_ld_path. However AFQ_mrtrix_set_ld_path takes as an input, the output of the present function. Gari, did this work for you on linux?

garikoitz replied:
Yes, I wrote a note. I don't have a problem with it in OSX, and in two
different linux boxes (both Red Hat), one normal linux box and the other
one part of a cluster. Are you using ubuntu?

This is the note:

% Jason, I don't have the AFQ_mrtrix_set_ld_path problem (in linux/OSX), I
get rid of it with

% the matlab call itself. Furthermore, it would be circular, since that
function

% is using the version information as well. We could us a function call
used in

% both versions. Please edit this as it is working for you and I
% will override it in my version if necessary.

I don't understand what your problem is, can you send me the error please?

On Tue, Jul 12, 2016 at 12:56 AM dstrodtman <notifications@github.com>
wrote:

> In utilities/check_mrTrix_Version.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70352285>:
>
> > @@ -0,0 +1,19 @@
> > +function mrTrixVersion = check_mrTrix_Version
>
> This doesn't look like it has been fixed. For this function to run
> properly we need to first call AFQ_mrtrix_set_ld_path. However
> AFQ_mrtrix_set_ld_path takes as an input, the output of the present
> function. Gari, did this work for you on linux?
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70352285>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEWQV5c4XAECeJhde8WXaJsfh1eYoxrYks5qUsoggaJpZM4JGbEU>
> .
>

garikoitz replied:
Furthermore, it doesn't make sense, because your previous function
check_mrtrix was doing the same but for many other functions. I am just
checking mrconvert. If it is not properly installed, it will fail, but it
will be the same for all the functions.

Check that you don't have a different problem, did you change your machine
recently?
I had a problem in linux with this as well, the call from the command line
would work but the call from matlab not, because matlab is using its own
libraries.
Can you send me the output of this please?
system('mrconvert -version')
system('gcc -v')

thanks!

On Tue, Jul 12, 2016 at 1:56 AM Garikoitz Lerma-Usabiaga <
garikoitz@gmail.com> wrote:

> Yes, I wrote a note. I don't have a problem with it in OSX, and in two
> different linux boxes (both Red Hat), one normal linux box and the other
> one part of a cluster. Are you using ubuntu?
>
> This is the note:
>
> % Jason, I don't have the AFQ_mrtrix_set_ld_path problem (in linux/OSX), I
> get rid of it with
>
> % the matlab call itself. Furthermore, it would be circular, since that
> function
>
> % is using the version information as well. We could us a function call
> used in
>
> % both versions. Please edit this as it is working for you and I
> % will override it in my version if necessary.
>
> I don't understand what your problem is, can you send me the error please?
>
> On Tue, Jul 12, 2016 at 12:56 AM dstrodtman <notifications@github.com>
> wrote:
>
>> In utilities/check_mrTrix_Version.m
>> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70352285>:
>>
>> > @@ -0,0 +1,19 @@
>> > +function mrTrixVersion = check_mrTrix_Version
>>
>> This doesn't look like it has been fixed. For this function to run
>> properly we need to first call AFQ_mrtrix_set_ld_path. However
>> AFQ_mrtrix_set_ld_path takes as an input, the output of the present
>> function. Gari, did this work for you on linux?
>>
>> —
>> You are receiving this because you modified the open/close state.
>>
>>
>> Reply to this email directly, view it on GitHub
>> <https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70352285>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AEWQV5c4XAECeJhde8WXaJsfh1eYoxrYks5qUsoggaJpZM4JGbEU>
>> .
>>
>

jyeatman replied:
Without this call to AFQ_mrtrix_set_ld_path it is impossible to make any
system call to mrtrix. So this means that any function to check if mrtrix
is installed fails.

Jason D. Yeatman, PhD
Assistant Professor, Institute for Learning & Brain Sciences (I-LABS)
Department of Speech & Hearing Sciences
University of Washington
http://brainandeducation.com

On Mon, Jul 11, 2016 at 4:57 PM, Garikoitz Lerma-Usabiaga <
notifications@github.com> wrote:

> In utilities/check_mrTrix_Version.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70358150>:
>
> > @@ -0,0 +1,19 @@
> > +function mrTrixVersion = check_mrTrix_Version
>
> Yes, I wrote a note. I don't have a problem with it in OSX, and in two
> different linux boxes (both Red Hat), one normal linux box and the other
> one part of a cluster. Are you using ubuntu? This is the note: % Jason, I
> don't have the AFQ_mrtrix_set_ld_path problem (in linux/OSX), I get rid of
> it with % the matlab call itself. Furthermore, it would be circular, since
> that function % is using the version information as well. We could us a
> function call used in % both versions. Please edit this as it is working
> for you and I % will override it in my version if necessary. I don't
> understand what your problem is, can you send me the error please?
> … <#m_5681299657681076971_>
> On Tue, Jul 12, 2016 at 12:56 AM dstrodtman ***@***.***> wrote: In
> utilities/check_mrTrix_Version.m <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70352285>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version This doesn't
> look like it has been fixed. For this function to run properly we need to
> first call AFQ_mrtrix_set_ld_path. However AFQ_mrtrix_set_ld_path takes as
> an input, the output of the present function. Gari, did this work for you
> on linux? — You are receiving this because you modified the open/close
> state. Reply to this email directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70352285>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AEWQV5c4XAECeJhde8WXaJsfh1eYoxrYks5qUsoggaJpZM4JGbEU>
> .
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70358150>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABsYyh8UDRjPcpSV02PrCA23b81jpIUWks5qUthSgaJpZM4JGbEU>
> .
>

garikoitz replied:
hmmmm not in my case...
Can you please send me the outputs of both system calls I sent you before?

system('mrconvert -version')
system('gcc -v')

On Tue, Jul 12, 2016 at 2:07 AM Jason D. Yeatman <notifications@github.com>
wrote:

> In utilities/check_mrTrix_Version.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70359065>:
>
> > @@ -0,0 +1,19 @@
> > +function mrTrixVersion = check_mrTrix_Version
>
> Without this call to AFQ_mrtrix_set_ld_path it is impossible to make any
> system call to mrtrix. So this means that any function to check if mrtrix
> is installed fails. Jason D. Yeatman, PhD Assistant Professor, Institute
> for Learning & Brain Sciences (I-LABS) Department of Speech & Hearing
> Sciences University of Washington http://brainandeducation.com On Mon,
> Jul 11, 2016 at 4:57 PM, Garikoitz Lerma-Usabiaga <
> notifications@github.com> wrote:
> … <#m_8673309125022752045_>
> In utilities/check_mrTrix_Version.m <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70358150>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version Yes, I wrote
> a note. I don't have a problem with it in OSX, and in two different linux
> boxes (both Red Hat), one normal linux box and the other one part of a
> cluster. Are you using ubuntu? This is the note: % Jason, I don't have the
> AFQ_mrtrix_set_ld_path problem (in linux/OSX), I get rid of it with % the
> matlab call itself. Furthermore, it would be circular, since that function
> % is using the version information as well. We could us a function call
> used in % both versions. Please edit this as it is working for you and I %
> will override it in my version if necessary. I don't understand what your
> problem is, can you send me the error please? … <#m_5681299657681076971_>
> On Tue, Jul 12, 2016 at 12:56 AM dstrodtman ***@***.***> wrote: In
> utilities/check_mrTrix_Version.m <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> (comment) <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70352285>>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version This doesn't
> look like it has been fixed. For this function to run properly we need to
> first call AFQ_mrtrix_set_ld_path. However AFQ_mrtrix_set_ld_path takes as
> an input, the output of the present function. Gari, did this work for you
> on linux? — You are receiving this because you modified the open/close
> state. Reply to this email directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70352285>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AEWQV5c4XAECeJhde8WXaJsfh1eYoxrYks5qUsoggaJpZM4JGbEU>
> . — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70358150>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/ABsYyh8UDRjPcpSV02PrCA23b81jpIUWks5qUthSgaJpZM4JGbEU>
> .
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70359065>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEWQV4M9A7NC_D10jp3vaSKStrAakiByks5qUtrcgaJpZM4JGbEU>
> .
>

jyeatman replied:
In the previous versions I just used
which
to see if the functions were installed in the system. This will work fine
But without resetting the library paths none of these functions will run.
In your new version you check the output of mrview to get the version
number. mrview will not give the correct output on an ubuntu system without
setting the library paths properly.

One solution would be to check the mrtrix version without calling mrview.
For example just use system('which ') with the name of an mrtrix3 specific
function.

Jason D. Yeatman, PhD
Assistant Professor, Institute for Learning & Brain Sciences (I-LABS)
Department of Speech & Hearing Sciences
University of Washington
http://brainandeducation.com

On Mon, Jul 11, 2016 at 5:18 PM, Garikoitz Lerma-Usabiaga <
notifications@github.com> wrote:

> In utilities/check_mrTrix_Version.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70359943>:
>
> > @@ -0,0 +1,19 @@
> > +function mrTrixVersion = check_mrTrix_Version
>
> hmmmm not in my case... Can you please send me the outputs of both system
> calls I sent you before? system('mrconvert -version') system('gcc -v') On
> Tue, Jul 12, 2016 at 2:07 AM Jason D. Yeatman <notifications@github.com>
> wrote:
> … <#m_-573941099128601710_>
> In utilities/check_mrTrix_Version.m <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70359065>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version Without this
> call to AFQ_mrtrix_set_ld_path it is impossible to make any system call to
> mrtrix. So this means that any function to check if mrtrix is installed
> fails. Jason D. Yeatman, PhD Assistant Professor, Institute for Learning &
> Brain Sciences (I-LABS) Department of Speech & Hearing Sciences University
> of Washington http://brainandeducation.com On Mon, Jul 11, 2016 at 4:57
> PM, Garikoitz Lerma-Usabiaga < ***@***.***> wrote: …
> <#m_8673309125022752045_> In utilities/check_mrTrix_Version.m <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> (comment) <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70358150>>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version Yes, I wrote
> a note. I don't have a problem with it in OSX, and in two different linux
> boxes (both Red Hat), one normal linux box and the other one part of a
> cluster. Are you using ubuntu? This is the note: % Jason, I don't have the
> AFQ_mrtrix_set_ld_path problem (in linux/OSX), I get rid of it with % the
> matlab call itself. Furthermore, it would be circular, since that function
> % is using the version information as well. We could us a function call
> used in % both versions. Please edit this as it is working for you and I %
> will override it in my version if necessary. I don't understand what your
> problem is, can you send me the error please? … <#m_5681299657681076971_>
> On Tue, Jul 12, 2016 at 12:56 AM dstrodtman ***@***.***> wrote: In
> utilities/check_mrTrix_Version.m <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> <#11
> <https://github.com/yeatmanlab/AFQ/pull/11>> (comment) <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> (comment) <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70352285>>>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version This doesn't
> look like it has been fixed. For this function to run properly we need to
> first call AFQ_mrtrix_set_ld_path. However AFQ_mrtrix_set_ld_path takes as
> an input, the output of the present function. Gari, did this work for you
> on linux? — You are receiving this because you modified the open/close
> state. Reply to this email directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70352285>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AEWQV5c4XAECeJhde8WXaJsfh1eYoxrYks5qUsoggaJpZM4JGbEU>
> . — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70358150>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/ABsYyh8UDRjPcpSV02PrCA23b81jpIUWks5qUthSgaJpZM4JGbEU>
> . — You are receiving this because you modified the open/close state. Reply
> to this email directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70359065>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AEWQV4M9A7NC_D10jp3vaSKStrAakiByks5qUtrcgaJpZM4JGbEU>
> .
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70359943>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABsYyuFUfpYiJZEKkmZNckS1LKtpFkptks5qUt1ggaJpZM4JGbEU>
> .
>

garikoitz replied:
ok, I'll change and commit.

What do you think about this? You just launch your matlab with this command:
LD_PRELOAD="/usr/local/eb/software/GCC/4.9.3/lib64/libstdc++.so.6" matlab &
you can even create an alias in bash_profile:
alias matlab_ld='LD_PRELOAD="/usr/local/eb/software/GCC/4.9.3/lib64/libstdc++.so.6"
matlab &'

in any case, I'll edit it and commit it.

On Tue, Jul 12, 2016 at 2:48 AM Jason D. Yeatman <notifications@github.com>
wrote:

> In utilities/check_mrTrix_Version.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70362495>:
>
> > @@ -0,0 +1,19 @@
> > +function mrTrixVersion = check_mrTrix_Version
>
> In the previous versions I just used which to see if the functions were
> installed in the system. This will work fine But without resetting the
> library paths none of these functions will run. In your new version you
> check the output of mrview to get the version number. mrview will not give
> the correct output on an ubuntu system without setting the library paths
> properly. One solution would be to check the mrtrix version without calling
> mrview. For example just use system('which ') with the name of an mrtrix3
> specific function. Jason D. Yeatman, PhD Assistant Professor, Institute for
> Learning & Brain Sciences (I-LABS) Department of Speech & Hearing Sciences
> University of Washington http://brainandeducation.com On Mon, Jul 11,
> 2016 at 5:18 PM, Garikoitz Lerma-Usabiaga < notifications@github.com>
> wrote:
> … <#m_3342909447770579329_>
> In utilities/check_mrTrix_Version.m <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70359943>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version hmmmm not in
> my case... Can you please send me the outputs of both system calls I sent
> you before? system('mrconvert -version') system('gcc -v') On Tue, Jul 12,
> 2016 at 2:07 AM Jason D. Yeatman ***@***.***> wrote: …
> <#m_-573941099128601710_> In utilities/check_mrTrix_Version.m <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> (comment) <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70359065>>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version Without this
> call to AFQ_mrtrix_set_ld_path it is impossible to make any system call to
> mrtrix. So this means that any function to check if mrtrix is installed
> fails. Jason D. Yeatman, PhD Assistant Professor, Institute for Learning &
> Brain Sciences (I-LABS) Department of Speech & Hearing Sciences University
> of Washington http://brainandeducation.com On Mon, Jul 11, 2016 at 4:57
> PM, Garikoitz Lerma-Usabiaga < ***@***.***> wrote: …
> <#m_8673309125022752045_> In utilities/check_mrTrix_Version.m <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> <#11
> <https://github.com/yeatmanlab/AFQ/pull/11>> (comment) <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> (comment) <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70358150>>>>: > @@
> -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version Yes, I wrote
> a note. I don't have a problem with it in OSX, and in two different linux
> boxes (both Red Hat), one normal linux box and the other one part of a
> cluster. Are you using ubuntu? This is the note: % Jason, I don't have the
> AFQ_mrtrix_set_ld_path problem (in linux/OSX), I get rid of it with % the
> matlab call itself. Furthermore, it would be circular, since that function
> % is using the version information as well. We could us a function call
> used in % both versions. Please edit this as it is working for you and I %
> will override it in my version if necessary. I don't understand what your
> problem is, can you send me the error please? … <#m_5681299657681076971_>
> On Tue, Jul 12, 2016 at 12:56 AM dstrodtman ***@***.***> wrote: In
> utilities/check_mrTrix_Version.m <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> <#11
> <https://github.com/yeatmanlab/AFQ/pull/11>> <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> <#11
> <https://github.com/yeatmanlab/AFQ/pull/11>>> (comment) <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> <#11
> <https://github.com/yeatmanlab/AFQ/pull/11>> (comment) <#11
> <https://github.com/yeatmanlab/AFQ/pull/11> (comment) <#11 (comment)
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70352285>>>>>: >
> @@ -0,0 +1,19 @@ > +function mrTrixVersion = check_mrTrix_Version This
> doesn't look like it has been fixed. For this function to run properly we
> need to first call AFQ_mrtrix_set_ld_path. However AFQ_mrtrix_set_ld_path
> takes as an input, the output of the present function. Gari, did this work
> for you on linux? — You are receiving this because you modified the
> open/close state. Reply to this email directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70352285>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AEWQV5c4XAECeJhde8WXaJsfh1eYoxrYks5qUsoggaJpZM4JGbEU>
> . — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70358150>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/ABsYyh8UDRjPcpSV02PrCA23b81jpIUWks5qUthSgaJpZM4JGbEU>
> . — You are receiving this because you modified the open/close state. Reply
> to this email directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70359065>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AEWQV4M9A7NC_D10jp3vaSKStrAakiByks5qUtrcgaJpZM4JGbEU>
> . — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub <
> https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70359943>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/ABsYyuFUfpYiJZEKkmZNckS1LKtpFkptks5qUt1ggaJpZM4JGbEU>
> .
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r70362495>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEWQVz5fY5xOHlMjeEuEIVQetI74n-8Wks5qUuRJgaJpZM4JGbEU>
> .
>

garikoitz replied:
Just committed the modified version, please check.
By the way, this doesn't fix it in Red Hat, I need the solution I sent before...

jyeatman commented on utilities/AFQ_mrtrixInit.m:
```diff
@@ -1,4 +1,4 @@
-function files = AFQ_mrtrixInit(dt6, lmax, mrtrix_folder)
+function files = AFQ_mrtrixInit(dt6, lmax, mrtrix_folder, mrtrixVersion)
```
I haven't sorted through all this code but files are being written out with only the _dt.mif _sf.mif and so forth. They are being written out in the right location, but we are losing some aspect of the file name. @garikoitz can you check why that might be?

garikoitz replied:
OK, I will check it in the server instead of my local osx machine. Are you going to make changes to the other files or should I commit everything back?

jyeatman replied:
I think it makes the most sense for you to fix these changes in your PR.
Then I will check them again. If I start editing code at the same time we
are likely to end up with conflicting versions :-)

Jason D. Yeatman, PhD
Assistant Professor, Institute for Learning & Brain Sciences (I-LABS)
Department of Speech & Hearing Sciences
University of Washington
http://brainandeducation.com

On Thu, Jul 7, 2016 at 2:25 PM, Garikoitz Lerma-Usabiaga <
notifications@github.com> wrote:

> In utilities/AFQ_mrtrixInit.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r69990118>:
>
> > @@ -1,4 +1,4 @@
> > -function files = AFQ_mrtrixInit(dt6, lmax, mrtrix_folder)
> > +function files = AFQ_mrtrixInit(dt6, lmax, mrtrix_folder, mrtrixVersion)
>
> OK, I will check it in the server instead of my local osx machine. Are you
> going to make changes to the other files or should I commit everything back?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/7fc540e2aad16177aeded65d4e6516e6f78cf07e#r69990118>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABsYynqxc9DMa2d9u11BQiOc2lIAALa2ks5qTW6sgaJpZM4JGbEU>
> .
>

jyeatman commented on functions/AFQ_WholebrainTractography.m:
```diff
@@ -77,7 +80,15 @@
+                                                      mrtrixpaths.wm, ...
+                                                      mrtrixpaths.wm,...
+                                                      'stream', ...
+                                                      opts.nfibers,...
```
Lets add a new option into the afq structure that sets the tractography algorithm used here. We can make SD_Stream the default but allow users a choice

garikoitz replied:
ok

jyeatman commented:
I was just looking through this to follow the progress. Let me know when you think it is ready to be checked and merged

garikoitz commented:
Jason, I think it is ok to check it again. The issues in the previous one have been addressed and now I have added the msmt options as well. You will have to change your preprocessing call in order to create concatenated files. There are some issues open to comment, if you want we can Skype right now.

garikoitz commented:
Hi ,if you haven't started yet wait a little bit until the next commit, I am adding freesurfer to the 5ttgen options.

dstrodtman commented:
Hi Gari,

The specific error that I'm getting from check_mrTrix_Version is the call
 [status, cmdout] = system('mrconvert -version');
cmdout = 

mrconvert: /usr/local/MATLAB/R2014a/sys/os/glnxa64/libstdc++.so.6: version `GLIBCXX_3.4.19' not found (required by mrconvert)
mrconvert: /usr/local/MATLAB/R2014a/sys/os/glnxa64/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by /home/dstrodtman/git/mrtrix3/release/bin/../lib/libmrtrix-0.3.15.so)
mrconvert: /usr/local/MATLAB/R2014a/sys/os/glnxa64/libstdc++.so.6: version `GLIBCXX_3.4.19' not found (required by /home/dstrodtman/git/mrtrix3/release/bin/../lib/libmrtrix-0.3.15.so)

It also appears that in AFQ_mrtrix_cmd it's defaulting to 5ttgen, which has broken the workaround that I had used with you code last Friday. The error being:
Error: unrecognized arguments: -nthreads 0

usage: 5ttgen [-continue <TempDir> <LastFile>] [-force] [-help] [-nocleanup]
              [-nthreads number] [-tempdir /path/to/tmp/] [-quiet | -verbose]
              {fsl,freesurfer} ...

This leads to a string of further errors, which I'm happy to share, but I'm hoping that your progress today will take care of these. Is it necessary to default to the 5ttgen? Do you know why the previous version of your code might have worked--has this option been added since then?

Thanks for all your work on this--let me know if there's a specific condition under which you'd like me to test this.

garikoitz commented:
Thanks!
Yes, the problem you have is the same one I had. Please see the other
comments I did yesterday, with the Matlab call I posted this would be
solved. In any case, as per Jason's suggestion, I already changed the
check_mrTrix_Version and used only system('which '-s, so now it should work
as before for you. I committed it yesterday.

About the 5ttgen: you need it if you have multishell data. The -nthreads 0
is working for me, that's strange. The problem I have now is with the /tmp
folders. I am trying to troubleshoot it, but it seems something that they
have to fix in their cpp code. The problem is that even if you specify to
use your own /tmp folder, this parameter is not passed to the subsequent
script calls, and those are the ones failing. This works fine in OSX, but
in Linux in the cluster is not working. In the process I am testing several
things, so maybe is better if you wait a little bit, I will comment it with
the next commit. As a summary: I only have problems now with the 5ttgen
script in Linux.

About the default: I added a param in AFQ_Create that makes it multishell =
true, and if multishell is true, then you have to do 5ttgen.

On Tue, Jul 12, 2016 at 8:32 PM dstrodtman <notifications@github.com> wrote:

> Hi Gari,
>
> The specific error that I'm getting from check_mrTrix_Version is the call
> [status, cmdout] = system('mrconvert -version');
> cmdout =
>
> mrconvert: /usr/local/MATLAB/R2014a/sys/os/glnxa64/libstdc++.so.6: version GLIBCXX_3.4.19'
> not found (required by mrconvert)
> mrconvert: /usr/local/MATLAB/R2014a/sys/os/glnxa64/libstdc++.so.6: versionGLIBCXX_3.4.18'
> not found (required by /home/dstrodtman/git/mrtrix3/release/bin/../lib/
> libmrtrix-0.3.15.so)
> mrconvert: /usr/local/MATLAB/R2014a/sys/os/glnxa64/libstdc++.so.6: version
> `GLIBCXX_3.4.19' not found (required by
> /home/dstrodtman/git/mrtrix3/release/bin/../lib/libmrtrix-0.3.15.so)
>
> It also appears that in AFQ_mrtrix_cmd it's defaulting to 5ttgen, which
> has broken the workaround that I had used with you code last Friday. The
> error being:
> Error: unrecognized arguments: -nthreads 0
>
> usage: 5ttgen [-continue ] [-force] [-help] [-nocleanup]
> [-nthreads number] [-tempdir /path/to/tmp/] [-quiet | -verbose]
> {fsl,freesurfer} ...
>
> This leads to a string of further errors, which I'm happy to share, but
> I'm hoping that your progress today will take care of these. Is it
> necessary to default to the 5ttgen? Do you know why the previous version of
> your code might have worked--has this option been added since then?
>
> Thanks for all your work on this--let me know if there's a specific
> condition under which you'd like me to test this.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-232137751>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AEWQV2RE-rwLccmLBs6JFLRGKp8ij61Qks5qU93MgaJpZM4JGbEU>
> .
>

dstrodtman commented:
Adding the multishell as an optional varargin will make your code backwards compatible with everyone else's processing pipeline.

And I realized after my previous post that I've not reinstalled Freesurfer since I rebuilt my OS. I'll do that now. But given the code that you currently have written, I can bypass this by making multishell = 0 in AFQ_Create, given that I have single shell data? Testing now.

dstrodtman commented:
Setting multishell = false, I get the following error. 

Reference to non-existent field 'wmCsd'.
Error in AFQ_Create (line 303)
            afq.files.mrtrix.csd{ii} = files.wmCsd;

dstrodtman commented on functions/AFQ_Create.m:
```diff
@@ -268,11 +290,26 @@
-        files = mrtrix_init(AFQ_get(afq, 'dt6 path',ii),lmax,mrtrixdir);
-        afq.files.mrtrix.csd{ii} = files.csd;
-        afq.files.mrtrix.wm{ii} = files.wm;
+        multishell = true;
```
This overrides setting multishell to false earlier in your code.

garikoitz replied:
Changed. Thanks

dstrodtman commented on functions/AFQ_Create.m:
```diff
@@ -268,11 +290,26 @@
+            afq.files.mrtrix.csd{ii} = files.csd;
+            afq.files.mrtrix.wm{ii} = files.wmMask;
+        else
+            afq.files.mrtrix.csd{ii} = files.wmCsd;
```
How does this exist? Aren't we using mrTrix to generate the csd?

garikoitz replied:
Yes, let me finish with the revision and I will update. 
In not-multishell wm was the wm mask, but in multishell it is the csd of the wm. 
I was changing the names in order to make them as similiar as the previous version as possible, but it is going to be a mess. 
As I told this afternoon, please wait until I finish again, it is being impossible to make 5ttgen work in linux and I have to edit mrtrix python code now. 
Sorry about the delay.

dstrodtman commented on functions/AFQ_Create.m:
```diff
@@ -188,8 +193,25 @@
+%                         Seedtest, Tensor_Det, Tensor_Prob (default: iFOD2).
+afq.params.track.mrTrixAlgo = 'iFOD2';
+% Specify here if you want multishell true or false
+afq.params.track.multishell = true;
```
This should default to false, or later code should be changed to not break other AFQ pipelines.

garikoitz replied:
Yes, you are right.

garikoitz commented:
@dstrodtman you can check AFQ_Create now if you want. Don't try with multishell yet please, I am changing AFQ_mrtrix_5ttgen right now, maybe I will change AFQ_mrtrixInit as well but as you said if multishell is 'false' it should be compatible.

jyeatman commented:
It would be best to split these pull requests to separate branches. It
makes it difficult for us to check anything if aspects of the code are
still changing

Jason D. Yeatman, PhD
Assistant Professor, Institute for Learning & Brain Sciences (I-LABS)
Department of Speech & Hearing Sciences
University of Washington
http://brainandeducation.com

On Tue, Jul 12, 2016 at 4:17 PM, Garikoitz Lerma-Usabiaga <
notifications@github.com> wrote:

> @dstrodtman <https://github.com/dstrodtman> you can check AFQ_Create now
> if you want. Don't try with multishell yet please, I am changing
> AFQ_mrtrix_5ttgen right now, maybe I will change AFQ_mrtrixInit as well but
> as you said if multishell is 'false' it should be compatible.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-232210420>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ABsYynTz8ac3yVFThmEAKba1nskyTJzKks5qVCB_gaJpZM4JGbEU>
> .
>

garikoitz commented:
Yes sorry, it is my very first pull request and it is too big and in a
rush. I will stop it now, in linux is working fine, in osx is working fine,
I am going to check that both multishell and not-multishell are working and
I will leave it like that.

In the cluster is failing a mrtrix system command, it is not going to
affect AFQ.

I will commit and write a comment soon.

On Wed, Jul 13, 2016 at 2:30 AM Jason D. Yeatman <notifications@github.com>
wrote:

> It would be best to split these pull requests to separate branches. It
> makes it difficult for us to check anything if aspects of the code are
> still changing
>
> Jason D. Yeatman, PhD
> Assistant Professor, Institute for Learning & Brain Sciences (I-LABS)
> Department of Speech & Hearing Sciences
> University of Washington
> http://brainandeducation.com
>
> On Tue, Jul 12, 2016 at 4:17 PM, Garikoitz Lerma-Usabiaga <
> notifications@github.com> wrote:
>
> > @dstrodtman <https://github.com/dstrodtman> you can check AFQ_Create now
> > if you want. Don't try with multishell yet please, I am changing
> > AFQ_mrtrix_5ttgen right now, maybe I will change AFQ_mrtrixInit as well
> but
> > as you said if multishell is 'false' it should be compatible.
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-232210420>, or
> mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe/ABsYynTz8ac3yVFThmEAKba1nskyTJzKks5qVCB_gaJpZM4JGbEU
> >
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-232221759>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AEWQV5C9beJ2OV4ldRdvrtW63B4cfFhzks5qVDGVgaJpZM4JGbEU>
> .
>

jyeatman commented:
Okay sounds good. You can put this pull request on a separate branch so
that you can keep developing while we go through and check all this

On Tuesday, July 12, 2016, Garikoitz Lerma-Usabiaga <
notifications@github.com> wrote:

> Yes sorry, it is my very first pull request and it is too big and in a
> rush. I will stop it now, in linux is working fine, in osx is working fine,
> I am going to check that both multishell and not-multishell are working and
> I will leave it like that.
>
> In the cluster is failing a mrtrix system command, it is not going to
> affect AFQ.
>
> I will commit and write a comment soon.
>
> On Wed, Jul 13, 2016 at 2:30 AM Jason D. Yeatman <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> wrote:
>
> > It would be best to split these pull requests to separate branches. It
> > makes it difficult for us to check anything if aspects of the code are
> > still changing
> >
> > Jason D. Yeatman, PhD
> > Assistant Professor, Institute for Learning & Brain Sciences (I-LABS)
> > Department of Speech & Hearing Sciences
> > University of Washington
> > http://brainandeducation.com
> >
> > On Tue, Jul 12, 2016 at 4:17 PM, Garikoitz Lerma-Usabiaga <
> > notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> >
> > > @dstrodtman <https://github.com/dstrodtman> you can check AFQ_Create
> now
> > > if you want. Don't try with multishell yet please, I am changing
> > > AFQ_mrtrix_5ttgen right now, maybe I will change AFQ_mrtrixInit as well
> > but
> > > as you said if multishell is 'false' it should be compatible.
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-232210420>, or
> > mute
> > > the thread
> > > <
> >
> https://github.com/notifications/unsubscribe/ABsYynTz8ac3yVFThmEAKba1nskyTJzKks5qVCB_gaJpZM4JGbEU
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-232221759>, or
> mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe/AEWQV5C9beJ2OV4ldRdvrtW63B4cfFhzks5qVDGVgaJpZM4JGbEU
> >
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-232222676>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ABsYymWavrSjFQDcISHrThGL3WKkGT4Hks5qVDL4gaJpZM4JGbEU>
> .
>


-- 
Jason D. Yeatman, PhD
Assistant Professor, Institute for Learning & Brain Sciences (I-LABS)
Department of Speech & Hearing Sciences
University of Washington
http://brainandeducation.com

garikoitz commented:
Thank you both for your patience. 
I have checked with several subjects from preprocessing until AFQ_Create and AFQ_Run, in OSX, in Linux and in Cluster Linux, with multishell = true and = false, and it is working fine for me. Multishell = false is the default.

There are notes in the code about the cluster issue. Basically it is related to how the /tmp folders are being used. I pass the -tempdir command now specifying which temp folder I want to use.

Please remember that there is some issues with the folders. Now I assume that it is creating in the proper place and with the proper prefix (at least for me), but as I don't know what is your setup I cannot check it. 

mrTrix2 hasn't been checked. 

Last question: Jason, do you usually use acpc-d T1w or the one from freesurfer without the AC-PC? It is going to be easier to plot in the surface and using the segmentations to use without the AC-PC right? If you use multishell with the 'freesurfer' option ('fsl' is the default), it will require aparc+aseg.mgz. We could rotate aparc+aseg.mgz to ac-pc as well with nearest neighbour (they are labels), but the results are not going to be very good. And then I don't know what the implications are for surfaces (do we need to register them as well?). What do you usually do in this regard? Right now I decided to work in FS space without ac-pc rotating and registering all DWI data to T1.mgz.

I won't touch the code anymore until you check it. 

Thanks again and please let me know if you find any issue.

dstrodtman commented:
Hi Gari,
I haven't had time to go through the results with Jason, but it does appear that processing a single shell is now working properly on my system. Thank you! Hopefully Jason and I will have time to run over the results soon and I can confirm that everything is in order. 

I tried to run multishell without reading your comments above and ran into that exact issue. I'm concerned about hard-coding a file that isn't generating during the AFQ process and hasn't previously been required to run the pipeline. If multishell processing requires Freesurfer and additional files, perhaps the best solution would be to have multishell == true trigger an initial check that Freesurfer is installed and that paths have been defined to these additional files. That way it will throw an error immediately if these conditions aren't met.

dstrodtman commented on utilities/AFQ_mrtrixInit.m:
```diff
@@ -80,101 +125,145 @@
+    % Create the 5tt file from the same ac-pc-ed T1 nii we used in the other steps: 
+    if (~computed.('tt5')) && (mrtrixVersion > 2)
+
+        tool = 'freesurfer'; % Options are 'fsl' or 'freesurfer'
```
Can tool be set without directly modifying the code?

dstrodtman commented on utilities/AFQ_mrtrixInit.m:
```diff
@@ -80,101 +125,145 @@
+                error(['Cannot find T1, please copy it to ' session]);
+            end              
+        else
+            inputFile = fullfile(session,'aparc+aseg.mgz');
```
I'm starting with preprocessed Human Connectome Project data, and have aparc+aseg.nii.gz instead in the location specified. Should this file work in place of the file you specified, if the code were modified to be more robust? I'm looking through mrTrix3 documentation now.

garikoitz replied:
Yes, I will modify it to accept mgz or nii or nii.gz.
I can add 'tool' as an afq parameter in AFQ_Create and specify that you
will need apart+aseg (or any 'aseg') in order to make it work.
I will make 'fsl' as the default, as it doesn't require extra files (but
for me the results are far worse...)
Regarding the ac-pc question: I understand that you are inputing a T1 that
hasn't been ac-pc-ed so all your DWI pipeline is going to be in freesurfer
space, is that right?

On Thu, Jul 14, 2016 at 2:13 AM dstrodtman <notifications@github.com> wrote:

> In utilities/AFQ_mrtrixInit.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70730587>:
>
> > +                          mrtrixVersion)
> > +    end
> > +
> > +else
> > +    % Create the 5tt file from the same ac-pc-ed T1 nii we used in the other steps:
> > +    if (~computed.('tt5')) && (mrtrixVersion > 2)
> > +
> > +        tool = 'freesurfer'; % Options are 'fsl' or 'freesurfer'
> > +
> > +        if strcmp(tool, 'fsl')
> > +            inputFile = fullfile(session, dt_info.files.t1);
> > +            if ~(exist(inputFile, 'file') == 2)
> > +                error(['Cannot find T1, please copy it to ' session]);
> > +            end
> > +        else
> > +            inputFile = fullfile(session,'aparc+aseg.mgz');
>
> I'm starting with preprocessed Human Connectome Project data, and have
> aparc+aseg.nii.gz instead in the location specified. Should this file work
> in place of the file you specified, if the code were modified to be more
> robust? I'm looking through mrTrix3 documentation now.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/04eb7b672913be5a6d822faf4964d7e59274eb25#r70730587>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEWQVzGuEdWwLpDgf-fJu5P2tr5UUUrpks5qVX8hgaJpZM4JGbEU>
> .
>

dstrodtman commented on utilities/AFQ_mrtrixInit.m:
```diff
@@ -80,101 +126,154 @@
+           asegFiles = dir(fullfile(session,'*aseg*'));
+           for ii = 1:length(asegFiles)
+               if length(strfind(asegFiles(ii).name, 'aseg')) > 0
+                   inputFile = asegFiles(ii).name;
```
Need to change all these to absolute paths
 inputFile = fullfile(session, asegFiles(ii).name);

garikoitz replied:
session is the full path for me. That's the part I commented that I don't
know your exact dir structure and how you make the calls. I use a modified
bde_preprocessdiffusion.m script with full path names. Can you re-check
this again please? Otherwise make it the way it works for you and I will
adapt it for me afterwards if you think it will be easier... that do you
think?

On Fri, Jul 15, 2016 at 7:04 PM dstrodtman <notifications@github.com> wrote:

> In utilities/AFQ_mrtrixInit.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r71006066>:
>
> > +    % Create the 5tt file from the same ac-pc-ed T1 nii we used in the other steps:
> > +    if (~computed.('tt5')) && (mrtrixVersion > 2)
> > +        inputFile = [];
> > +        if strcmp(tool, 'fsl')
> > +            inputFile = fullfile(session, dt_info.files.t1);
> > +            if ~(exist(inputFile, 'file') == 2)
> > +                error(['Cannot find T1, please copy it to ' session]);
> > +            end
> > +        % Find and aseg file and better if it is an aparc+aseg one.
> > +        % Select the first aparc if there are several *aseg* files.
> > +        % It can take mgz or nii or mif
> > +        else
> > +           asegFiles = dir(fullfile(session,'*aseg*'));
> > +           for ii = 1:length(asegFiles)
> > +               if length(strfind(asegFiles(ii).name, 'aseg')) > 0
> > +                   inputFile = asegFiles(ii).name;
>
> Need to change all these to absolute paths
> inputFile = fullfile(session, asegFiles(ii).name);
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/e6e990255c225454a1920d786c05a6a68c920d07#r71006066>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEWQV_s3eQn3FfUpdN79Diho0JatGZXjks5qV72kgaJpZM4JGbEU>
> .
>

dstrodtman replied:
The issue here is that the dir command strips the directory from session. The *.name that results from a dir command is only the file name. I'm surprised that this is working for you at all?

dstrodtman commented on utilities/AFQ_mrtrixInit.m:
```diff
@@ -80,101 +126,154 @@
+                   inputFile = asegFiles(ii).name;
+               end
+           end
+            if ~(exist(inputFile, 'file') == 2)
```
Without absolute paths this throws an error. We should not assume that AFQ is being run with the session directory as the current directory, as this is not required for default operation.

garikoitz replied:
Yes, that's what I am saying, in my case session has an absolute path, what
is your input to AFQ_Create?

On Fri, Jul 15, 2016 at 7:58 PM dstrodtman <notifications@github.com> wrote:

> In utilities/AFQ_mrtrixInit.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r71014852>:
>
> > +                error(['Cannot find T1, please copy it to ' session]);
> > +            end
> > +        % Find and aseg file and better if it is an aparc+aseg one.
> > +        % Select the first aparc if there are several *aseg* files.
> > +        % It can take mgz or nii or mif
> > +        else
> > +           asegFiles = dir(fullfile(session,'*aseg*'));
> > +           for ii = 1:length(asegFiles)
> > +               if length(strfind(asegFiles(ii).
>
> name, 'aseg')) > 0
> > +                   inputFile = asegFiles(ii).name;
> > +               end
> > +               if length(strfind(asegFiles(ii).name, 'aparc')) > 0
> > +                   inputFile = asegFiles(ii).name;
> > +               end
> > +           end
> > +            if ~(exist(inputFile, 'file') == 2)
>
> Without absolute paths this throws an error. We should not assume that AFQ
> is being run with the session directory as the current directory, as this
> is not required for default operation.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/e6e990255c225454a1920d786c05a6a68c920d07#r71014852>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEWQVydz8W4X_-H7r3wNnossjyWhAfu9ks5qV8pFgaJpZM4JGbEU>
> .
>

dstrodtman replied:
Yes, session at this point has an absolute path. I call 'sub_dirs' with a list of full paths to my dti*trilin directory for each subject when I AFQ_Create.

garikoitz replied:
ok! but it is working for you then? Should I check sthg else? I can do it
in 3-4 for hours, I am leaving right now. Thanks for the testing!

On Fri, Jul 15, 2016 at 8:11 PM dstrodtman <notifications@github.com> wrote:

> In utilities/AFQ_mrtrixInit.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r71017035>:
>
> > +                error(['Cannot find T1, please copy it to ' session]);
> > +            end
> > +        % Find and aseg file and better if it is an aparc+aseg one.
> > +        % Select the first aparc if there are several *aseg* files.
> > +        % It can take mgz or nii or mif
> > +        else
> > +           asegFiles = dir(fullfile(session,'*aseg*'));
> > +           for ii = 1:length(asegFiles)
> > +               if length(strfind(asegFiles(ii).
>
> name, 'aseg')) > 0
> > +                   inputFile = asegFiles(ii).name;
> > +               end
> > +               if length(strfind(asegFiles(ii).name, 'aparc')) > 0
> > +                   inputFile = asegFiles(ii).name;
> > +               end
> > +           end
> > +            if ~(exist(inputFile, 'file') == 2)
>
> Yes, session at this point has an absolute path. I call 'sub_dirs' with a
> list of full paths to my dti*trilin directory for each subject when I
> AFQ_Create.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/e6e990255c225454a1920d786c05a6a68c920d07#r71017035>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEWQV-3oL-v_qTww-BDiUvqE8jE_FCs5ks5qV81qgaJpZM4JGbEU>
> .
>

dstrodtman commented on utilities/AFQ_mrtrix_5ttgen.m:
```diff
@@ -0,0 +1,46 @@
+               '-lut $FREESURFER_HOME/FreeSurferColorLUT.txt ' ...
+               input_file   ' ' ...
+               tt5_filename ' ' ...
+               '-nocrop -tempdir ' tmpDir];
```
This -tempdir call is throwing an error. The directory to which it points exists on my system. This option is just to physically write data rather than storing everything in RAM?

dstrodtman commented:
I'm going to be with subjects for the next few hours. I'll have a little time to spend later this afternoon on this, but our building's power is going down for the weekend so likely won't be able to do much until Monday at the earliest. Thanks for all your work on this!

garikoitz commented:
Yes, it was a quick and dirty solution for the /tmp problem I was having. I
have read some other reports online with the same problem. We can use any
folder name (~/AFQ_mrtrix_tmp). The thing is that ~/ should always be
writeable and there always should be space on disk. Can you change this
directly or should I change and commit it?

On Fri, Jul 15, 2016 at 9:20 PM dstrodtman <notifications@github.com> wrote:

> I'm going to be with subjects for the next few hours. I'll have a little
> time to spend later this afternoon on this, but our building's power is
> going down for the weekend so likely won't be able to do much until Monday
> at the earliest. Thanks for all your work on this!
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-233046603>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEWQVyBwkPfEByxOJPZxI7qW7An3XpcPks5qV91jgaJpZM4JGbEU>
> .
>

dstrodtman commented:
Gari,
You'd mentioned that you've been testing this on a compute cluster. AFQ_mrtrixInit is presently being called serially within a for loop in AFQ_Create. This is causing a huge bottleneck in my pipeline--does your set-up automatically work around this and allow for parallel processing of subjects, or are you just distributing the computation for a single subject?

garikoitz commented:
I launched every subject in parallel, I call a function based on
bde_preprocdiffusion for every subject in parallel.

On Sat, Jul 23, 2016 at 1:06 AM dstrodtman <notifications@github.com> wrote:

> Gari,
> You'd mentioned that you've been testing this on a compute cluster.
> AFQ_mrtrixInit is presently being called serially within a for loop in
> AFQ_Create. This is causing a huge bottleneck in my pipeline--does your
> set-up automatically work around this and allow for parallel processing of
> subjects, or are you just distributing the computation for a single subject?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11#issuecomment-234677921>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEWQV8r61HQRY-2eozlnqgcMbUmOHuiOks5qYUzcgaJpZM4JGbEU>
> .
>

jyeatman commented:
I am reorganizing all this in another PR
